modules/miner/miner.go

```diff
@@ -15,10 +15,19 @@ const (
 	iterationsPerAttempt = 16 * 1024
 
 	// headerForWorkMemory is the number of previous calls to 'headerForWork'
-	// that are remembered.
-	headerForWorkMemory = 50
+	// that are remembered. Additionally, 'headerForWork' will only poll for a
+	// new block every 'headerForWorkMemory / blockForWorkMemory' times it is
+	// called. This reduces the amount of memory used, but comes at the cost of
+	// not always having the most recent transactions
+	headerForWorkMemory = 1000
+
+	// blockForWorkMemory is the maximum number of blocks the miner will store
+	// Blocks take up to 2 megabytes of memory, so it is important to keep a cap
+	blockForWorkMemory = 50
```

Comment:
50 is too many. Blocks are big and I would like to keep Sia as lightweight as possible. I think 20 would be better.